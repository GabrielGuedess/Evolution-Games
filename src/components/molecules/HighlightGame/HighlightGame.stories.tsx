import { Story, Meta } from '@storybook/react/types-6-0';

import { HighlightGame, HighlightGameProps } from './HighlightGame';

export default {
  title: 'Molecules/HighlightGame',
  component: HighlightGame,
  args: {
    img: '/img/games/starWarsHighlight.png',
    gameLogo: '/img/games/highlightName/nameStarWars.png',
    gameLogoAlt: 'Logo Star Wars',
    alt: 'Star Wars',
    releaseYear: new Date(),
    genre: 'Action-adventure',
    developer: 'Respawn Entertainment',
    platform: ['pc', 'playstation', 'xbox'],
  },
} as Meta<HighlightGameProps>;

export const Default: Story<HighlightGameProps> = args => (
  <div style={{ width: '100%', height: '756px' }}>
    <HighlightGame {...args} />
  </div>
);
