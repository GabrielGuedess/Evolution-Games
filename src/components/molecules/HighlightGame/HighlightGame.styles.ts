import media from 'styled-media-query';

import styled, { css, keyframes } from 'styled-components';

const textWriting = keyframes`
  0% {
    background-position: -500%;
  }
  100% {
    background-position: 500%;
  }
`;

export const Wrapper = styled.section`
  position: relative;
  width: 100%;
  height: 100%;
`;

export const WrapperHighlight = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

export const WrapperInfo = styled.div`
  position: absolute;
  inset: 0;
  height: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

export const Info = styled.div`
  width: 60%;
  height: 100%;
  padding: 3rem 0;
  margin: 0 1.6rem 0 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  ${media.greaterThan('medium')`
    width: 33%;
    height: auto;
    margin: 0 10rem 0 0;
    justify-content: flex-start;
    gap: 5rem;
  `}
`;

export const Release = styled.p`
  ${({ theme }) => css`
    font-size: 1.2rem;
    line-height: 1.5rem;
    letter-spacing: 0.4rem;
    text-align: center;
    text-transform: uppercase;
    color: ${theme.colors.white};

    overflow: hidden;
    background: linear-gradient(90deg, #000, #fff, #fff);
    background-repeat: no-repeat;
    background-size: 80%;
    animation: ${textWriting} 2.5s linear infinite;
    animation-delay: 2s;
    -webkit-background-clip: text;
    -webkit-text-fill-color: rgba(255, 255, 255, 0);

    ${media.greaterThan('medium')`
      font-size: 2.4rem;
      line-height: 2.9rem;
    `}
  `}
`;

export const GenreDeveloper = styled.span`
  ${({ theme }) => css`
    font-size: 0.8rem;
    line-height: 0.8rem;
    color: ${theme.colors.secondary};

    ${media.greaterThan('medium')`
      font-size: 1.8rem;
      line-height: 2.2rem;
    `}
  `}
`;

export const Platforms = styled.div`
  display: flex;
  justify-content: center;
  gap: 3rem;
`;
