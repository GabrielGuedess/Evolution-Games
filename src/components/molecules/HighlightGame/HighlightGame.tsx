import Image from 'next/image';

import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';
import {
  Platform,
  PlatformIcon,
} from 'components/atoms/PlatformIcon/PlatformIcon';

import * as S from './HighlightGame.styles';

export type HighlightGameProps = {
  img: string;
  alt: string;
  gameLogo: string;
  gameLogoAlt: string;
  releaseYear: Date;
  genre: string;
  developer: string;
  platform: Platform[];
};

export const HighlightGame = ({
  img,
  alt,
  gameLogo,
  gameLogoAlt,
  releaseYear,
  genre,
  developer,
  platform,
}: HighlightGameProps) => (
  <S.Wrapper>
    <S.WrapperHighlight>
      <Image
        src={img}
        layout="fill"
        objectFit="cover"
        alt={alt}
        className="imageBackground"
        loading="lazy"
      />
    </S.WrapperHighlight>

    <S.WrapperInfo>
      <S.Info>
        <Image
          src={gameLogo}
          width={768}
          height={296}
          objectFit="contain"
          alt={gameLogoAlt}
          loading="lazy"
        />

        <S.Release>
          {releaseYear.toLocaleString('pt-BR', {
            day: 'numeric',
            month: 'long',
            year: 'numeric',
          })}
        </S.Release>
        <S.GenreDeveloper>
          {genre}, {developer}
        </S.GenreDeveloper>

        <MediaMatch lessThan="medium">
          <S.Platforms>
            {platform.map(item => (
              <PlatformIcon key={item} platform={item} size="small" />
            ))}
          </S.Platforms>
        </MediaMatch>

        <MediaMatch greaterThan="medium">
          <S.Platforms>
            {platform.map(item => (
              <PlatformIcon key={item} platform={item} />
            ))}
          </S.Platforms>
        </MediaMatch>
      </S.Info>
    </S.WrapperInfo>
  </S.Wrapper>
);
