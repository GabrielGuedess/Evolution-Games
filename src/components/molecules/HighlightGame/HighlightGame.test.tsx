import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import { HighlightGame, HighlightGameProps } from './HighlightGame';

const props: HighlightGameProps = {
  img: '/img/games/starWarsHighlight.png',
  gameLogo: '/img/games/highlightName/nameStarWars.png',
  gameLogoAlt: 'Logo Star Wars',
  alt: 'Star Wars',
  releaseYear: new Date('09-10-2022'),
  genre: 'Action-adventure',
  developer: 'Respawn Entertainment',
  platform: ['pc', 'playstation', 'xbox'],
};

describe('<HighlightGame />', () => {
  it('should render the heading', () => {
    // Arrange
    const { container } = renderWithProviders(<HighlightGame {...props} />);

    expect(screen.getByRole('img', { name: props.alt })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
