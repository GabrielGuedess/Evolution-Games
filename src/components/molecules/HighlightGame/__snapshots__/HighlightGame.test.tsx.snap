// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<HighlightGame /> should render the heading 1'] = `
.c6 {
  display: none;
}

.c11 {
  display: none;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FFFFFF;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c8:hover {
  cursor: pointer;
  color: #855aff;
}

.c9 {
  width: 2rem;
  height: 1rem;
}

.c10 {
  width: 1.8rem;
  height: 1.8rem;
}

.c12 {
  width: 3.2rem;
  height: 1.7rem;
}

.c13 {
  width: 2.8rem;
  height: 2.8rem;
}

.c14 {
  width: 2.2rem;
  height: 2.2rem;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c2 {
  position: absolute;
  inset: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 60%;
  height: 100%;
  padding: 3rem 0;
  margin: 0 1.6rem 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 1.2rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.4rem;
  -moz-letter-spacing: 0.4rem;
  -ms-letter-spacing: 0.4rem;
  letter-spacing: 0.4rem;
  text-align: center;
  text-transform: uppercase;
  color: #FFFFFF;
  overflow: hidden;
  background: linear-gradient(90deg,#000,#fff,#fff);
  background-repeat: no-repeat;
  background-size: 80%;
  -webkit-animation: jyhgBN 2.5s linear infinite;
  animation: jyhgBN 2.5s linear infinite;
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
  -webkit-background-clip: text;
  -webkit-text-fill-color: rgba(255,255,255,0);
}

.c5 {
  font-size: 0.8rem;
  line-height: 0.8rem;
  color: #717B8A;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 3rem;
}

@media (max-width:768px) {
  .c6 {
    display: block;
  }
}

@media (min-width:768px) {
  .c11 {
    display: block;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 33%;
    height: auto;
    margin: 0 10rem 0 0;
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
    gap: 5rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.4rem;
    line-height: 2.9rem;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 1.8rem;
    line-height: 2.2rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    >
      <img
        alt="Star Wars"
        class="imageBackground"
        data-nimg="fill"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
      />
      <noscript />
    </span>
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Logo Star Wars"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
      <p
        class="c4"
      >
        6 de outubro de 2022
      </p>
      <span
        class="c5"
      >
        Action-adventure
        , 
        Respawn Entertainment
      </span>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="PC Icon"
              class="c9"
              fill="none"
              role="figure"
              viewBox="0 0 34 19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.426 18 3.324.545h6.886c1.318 0 2.4.253 3.247.759.853.5 1.45 1.196 1.79 2.088.347.886.426 1.91.239 3.068-.193 1.165-.614 2.19-1.262 3.077a6.556 6.556 0 0 1-2.505 2.062c-1.029.495-2.216.742-3.563.742H3.767l.486-2.957h3.792c.716 0 1.327-.123 1.833-.367.511-.25.918-.594 1.219-1.031.3-.443.497-.952.588-1.526.096-.585.065-1.094-.094-1.525a1.87 1.87 0 0 0-.878-1.006c-.42-.244-.986-.366-1.696-.366H6.528L4.116 18H.426ZM33.232 6.656H29.55c.012-.488-.05-.92-.187-1.295a2.533 2.533 0 0 0-.614-.972 2.576 2.576 0 0 0-.98-.605 3.76 3.76 0 0 0-1.304-.213c-.932 0-1.787.233-2.565.699-.779.466-1.435 1.142-1.969 2.028-.534.881-.903 1.946-1.108 3.196-.199 1.205-.187 2.213.034 3.026.222.812.614 1.426 1.176 1.84.569.41 1.276.614 2.122.614a5.56 5.56 0 0 0 1.492-.196c.472-.136.9-.33 1.287-.58a4.309 4.309 0 0 0 1.023-.928c.295-.364.525-.776.69-1.236h3.707a8.442 8.442 0 0 1-1.09 2.318 8.73 8.73 0 0 1-1.816 1.995 8.76 8.76 0 0 1-2.463 1.38c-.92.341-1.929.512-3.026.512-1.585 0-2.945-.364-4.082-1.091-1.13-.727-1.946-1.776-2.446-3.145-.5-1.37-.594-3.02-.281-4.952.312-1.87.923-3.454 1.832-4.756.915-1.306 2.026-2.298 3.333-2.974a9.063 9.063 0 0 1 4.21-1.014c1.04 0 1.98.142 2.82.426.842.284 1.558.699 2.149 1.244a5.098 5.098 0 0 1 1.346 1.986c.301.784.432 1.682.392 2.693Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="PlayStation Icon"
              class="c10"
              fill="none"
              role="figure"
              viewBox="0 0 28 22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.567 16.039c-1.05.71-.7 2.057 1.538 2.697 2.309.781 4.826.994 7.276.569.139 0 .28-.072.35-.072v-2.415l-2.378.781c-.91.284-1.818.356-2.727.141-.7-.213-.561-.638.28-.994l4.826-1.703v-2.624l-6.717 2.342c-.839.284-1.68.71-2.448 1.278ZM16.796 5.393v6.883c2.868 1.42 5.106 0 5.106-3.69 0-3.76-1.329-5.463-5.177-6.81C14.697 1.062 12.598.425 10.5 0v20.51L15.396 22V4.755c0-.782 0-1.348.561-1.135.77.212.839.994.839 1.773Zm9.096 9.012c-2.028-.71-4.197-.994-6.296-.781-1.119.071-2.17.355-3.149.71l-.21.07v2.77l4.549-1.704c.909-.284 1.818-.356 2.727-.14.7.212.56.637-.28.993l-6.996 2.626v2.697l9.655-3.62c.7-.284 1.328-.637 1.889-1.206.49-.712.28-1.706-1.89-2.415Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="Xbox Icon"
              class="c10"
              fill="none"
              role="figure"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.348 9.664c1.736-2.564 3.408-4.06 3.408-4.06s-1.842-2.14-4.06-2.948l-.144-.036A9.786 9.786 0 0 0 .2 10c0 2.218.74 4.266 1.978 5.906 0-.192.026-3.076 2.17-6.242ZM19.8 10c0-2.944-1.3-5.582-3.352-7.38l-.14.04c-2.218.81-4.064 2.948-4.064 2.948s1.672 1.496 3.408 4.06c2.144 3.168 2.17 6.05 2.166 6.244A9.747 9.747 0 0 0 19.8 10ZM7.602 2.34c1.282.574 2.388 1.514 2.388 1.514s1.116-.936 2.398-1.514c1.61-.722 2.84-.494 3.164-.416A9.77 9.77 0 0 0 10 .2a9.77 9.77 0 0 0-5.552 1.724c.316-.078 1.54-.31 3.154.416ZM14.494 11.614c-2.026-2.488-4.502-4.108-4.502-4.108s-2.464 1.618-4.494 4.108c-1.742 2.14-2.388 3.71-2.738 4.716l-.056.21A9.769 9.769 0 0 0 10 19.8c2.9 0 5.504-1.26 7.298-3.26l-.062-.21c-.35-1.006-1.002-2.576-2.742-4.716Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="c11"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="PC Icon"
              class="c12"
              fill="none"
              role="figure"
              viewBox="0 0 34 19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.426 18 3.324.545h6.886c1.318 0 2.4.253 3.247.759.853.5 1.45 1.196 1.79 2.088.347.886.426 1.91.239 3.068-.193 1.165-.614 2.19-1.262 3.077a6.556 6.556 0 0 1-2.505 2.062c-1.029.495-2.216.742-3.563.742H3.767l.486-2.957h3.792c.716 0 1.327-.123 1.833-.367.511-.25.918-.594 1.219-1.031.3-.443.497-.952.588-1.526.096-.585.065-1.094-.094-1.525a1.87 1.87 0 0 0-.878-1.006c-.42-.244-.986-.366-1.696-.366H6.528L4.116 18H.426ZM33.232 6.656H29.55c.012-.488-.05-.92-.187-1.295a2.533 2.533 0 0 0-.614-.972 2.576 2.576 0 0 0-.98-.605 3.76 3.76 0 0 0-1.304-.213c-.932 0-1.787.233-2.565.699-.779.466-1.435 1.142-1.969 2.028-.534.881-.903 1.946-1.108 3.196-.199 1.205-.187 2.213.034 3.026.222.812.614 1.426 1.176 1.84.569.41 1.276.614 2.122.614a5.56 5.56 0 0 0 1.492-.196c.472-.136.9-.33 1.287-.58a4.309 4.309 0 0 0 1.023-.928c.295-.364.525-.776.69-1.236h3.707a8.442 8.442 0 0 1-1.09 2.318 8.73 8.73 0 0 1-1.816 1.995 8.76 8.76 0 0 1-2.463 1.38c-.92.341-1.929.512-3.026.512-1.585 0-2.945-.364-4.082-1.091-1.13-.727-1.946-1.776-2.446-3.145-.5-1.37-.594-3.02-.281-4.952.312-1.87.923-3.454 1.832-4.756.915-1.306 2.026-2.298 3.333-2.974a9.063 9.063 0 0 1 4.21-1.014c1.04 0 1.98.142 2.82.426.842.284 1.558.699 2.149 1.244a5.098 5.098 0 0 1 1.346 1.986c.301.784.432 1.682.392 2.693Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="PlayStation Icon"
              class="c13"
              fill="none"
              role="figure"
              viewBox="0 0 28 22"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.567 16.039c-1.05.71-.7 2.057 1.538 2.697 2.309.781 4.826.994 7.276.569.139 0 .28-.072.35-.072v-2.415l-2.378.781c-.91.284-1.818.356-2.727.141-.7-.213-.561-.638.28-.994l4.826-1.703v-2.624l-6.717 2.342c-.839.284-1.68.71-2.448 1.278ZM16.796 5.393v6.883c2.868 1.42 5.106 0 5.106-3.69 0-3.76-1.329-5.463-5.177-6.81C14.697 1.062 12.598.425 10.5 0v20.51L15.396 22V4.755c0-.782 0-1.348.561-1.135.77.212.839.994.839 1.773Zm9.096 9.012c-2.028-.71-4.197-.994-6.296-.781-1.119.071-2.17.355-3.149.71l-.21.07v2.77l4.549-1.704c.909-.284 1.818-.356 2.727-.14.7.212.56.637-.28.993l-6.996 2.626v2.697l9.655-3.62c.7-.284 1.328-.637 1.889-1.206.49-.712.28-1.706-1.89-2.415Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="c8"
            color="white"
          >
            <svg
              aria-label="Xbox Icon"
              class="c14"
              fill="none"
              role="figure"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.348 9.664c1.736-2.564 3.408-4.06 3.408-4.06s-1.842-2.14-4.06-2.948l-.144-.036A9.786 9.786 0 0 0 .2 10c0 2.218.74 4.266 1.978 5.906 0-.192.026-3.076 2.17-6.242ZM19.8 10c0-2.944-1.3-5.582-3.352-7.38l-.14.04c-2.218.81-4.064 2.948-4.064 2.948s1.672 1.496 3.408 4.06c2.144 3.168 2.17 6.05 2.166 6.244A9.747 9.747 0 0 0 19.8 10ZM7.602 2.34c1.282.574 2.388 1.514 2.388 1.514s1.116-.936 2.398-1.514c1.61-.722 2.84-.494 3.164-.416A9.77 9.77 0 0 0 10 .2a9.77 9.77 0 0 0-5.552 1.724c.316-.078 1.54-.31 3.154.416ZM14.494 11.614c-2.026-2.488-4.502-4.108-4.502-4.108s-2.464 1.618-4.494 4.108c-1.742 2.14-2.388 3.71-2.738 4.716l-.056.21A9.769 9.769 0 0 0 10 19.8c2.9 0 5.504-1.26 7.298-3.26l-.062-.21c-.35-1.006-1.002-2.576-2.742-4.716Z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
